openedx/features/discounts/tests/test_views.py

```diff
@@ -0,0 +1,37 @@
+"""Tests of openedx.features.discounts.views"""
+# -*- coding: utf-8 -*-
+import jwt
+
+from xmodule.modulestore.tests.factories import CourseFactory
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from student.tests.factories import UserFactory
+from rest_framework.test import APIRequestFactory
+
+from openedx.features.discounts.views import CourseUserDiscount
+
+
+class TestCourseUserDiscount(ModuleStoreTestCase):
+    """
+    CourseUserDiscount should return a jwt with the information if this combination of user and
+    course can receive a discount, and how much that discount should be.
+    """
+
+    def setUp(self):
+        super(TestCourseUserDiscount, self).setUp()
+        self.user = UserFactory.create()
+        self.course = CourseFactory.create(run='test', display_name='test')
+        self.request_factory = APIRequestFactory()
+
+    def test_course_user_discount(self):
+        """
+        Test that the api returns a jwt with the discount information
+        """
+        fake_request = self.request_factory.get('')
+        fake_request.user = self.user
+        response = CourseUserDiscount.as_view()(fake_request)
+        self.assertEqual(response.status_code, 200)
```

Comment:
You might get better results here by using the django test client. That will make sure that any configured middleware is automatically applied to the request.

Reply:
I'm not sure how to do this with setting the user in the request...Do you have an example you could direct me to?

Reply:
Use `client.login`, like:

https://github.com/edx/edx-platform/blob/810b8b60f81918985bb0a830d7c49440404d8595/common/djangoapps/student/tests/test_admin_views.py#L43

Reply:
I did, but I can't figure out why it doesn't work. Can you take a look?

Reply:
Fixed it.